traits-stubs/traits-stubs/trait_types.pyi

```diff
@@ -0,0 +1,426 @@
+class _BaseInt(_TraitType[_T, int]):
+    ...
+
+
+class BaseInt(_BaseInt[int]):
+    default_value: int = ...
+
+
+class Int(BaseInt):
+    default_value: int = ...
+
+
+# -----------------Float--------------------
+class _BaseFloat(_TraitType[_T, float]):
+    ...
+
+
+class BaseFloat(_BaseFloat[float]):
+    default_value: float = ...
+
+
+class Float(BaseFloat):
+    default_value: float = ...
+
+
+# -----------------Complex--------------------
+class _BaseComplex(_TraitType[_T, complex]):
+    ...
+
+
+class BaseComplex(_BaseComplex[complex]):
+    default_value: complex = ...
+
+
+class Complex(BaseComplex):
+    default_value: complex = ...
+
+
+# -----------------Str--------------------
+
+class _BaseStr(_TraitType[_T, str]):
+    default_value: str = ...
+
+
+class BaseStr(_BaseStr[str]):
+    default_value: str = ...
+
+
+class Str(BaseStr):
+    default_value: str = ...
+
+
+# -----------------Title--------------------
+
+class Title(Str):
+    default_value: str = ...
+
+
+# -----------------Bytes--------------------
+class _BaseBytes(_TraitType[_T, bytes]):
+    ...
+
+
+class BaseBytes(_BaseBytes[bytes]):
+    default_value: bytes = ...
+
+
+class Bytes(BaseBytes):
+    default_value: bytes = ...
+
+
+# -----------------Bool--------------------
+class _BaseBool(_TraitType[_T, bool]):
+    ...
+
+
+class BaseBool(_BaseBool[bool]):
+    default_value: bool = ...
+
+
+class Bool(BaseBool):
+    default_value: bool = ...
+
+
+# -----------------BaseCInt--------------------
+
+class BaseCInt(BaseInt):
+    ...
+
+
+class CInt(BaseCInt):
+    ...
+
+
+# -----------------BaseCFloat--------------------
+
+class BaseCFloat(BaseFloat):
+    ...
+
+
+class CFloat(BaseCFloat):
+    ...
+
+
+# -----------------BaseCComplex--------------------
+
+class BaseCComplex(BaseComplex):
+    ...
+
+
+class CComplex(BaseCComplex):
+    ...
+
+
+# -----------------BaseCStr--------------------
+
+class BaseCStr(BaseStr):
+    ...
+
+
+class CStr(BaseCStr):
+    ...
+
+
+# -----------------BaseCBytes--------------------
+
+class BaseCBytes(BaseBytes):
+    ...
+
+
+class CBytes(BaseCBytes):
+    ...
+
+
+# -----------------BaseCBool--------------------
+
+class BaseCBool(BaseBool):
+    ...
+
+
+class CBool(BaseCBool):
+    ...
+
+
+# -----------------String--------------------
+
+class _String(_TraitType[_T, str]):
+    ...
+
+
+class String(_String[str]):
+    def __init__(self, value: str = ..., minlen: int = ..., maxlen: int = ...,
+                 regex: str = ...):
+        ...
+
+
+# -----------------Regex--------------------
+
+class Regex(String):
+    ...
+
+
+# -----------------Code--------------------
+
+class Code(String):
+    ...
+
+
+# -----------------HTML--------------------
+
+class HTML(String):
+    ...
+
+
+# -----------------Password--------------------
+
+class Password(String):
+    ...
+
+
+# -----------------Callable--------------------
+
+class _BaseCallable(_TraitType[_T, _CallableType[..., _Any]]):
```

Comment:
`None` is an allowed value so you want to use `typing`'s `Optional`; so something like:
```
from typing import Optional as _Optional
...
_OptionalCallable = _Optional[_CallableType[..., _Any]]

class BaseCallable(_TraitType[_OptionalCallable, _OptionalCallable]):
    ...
```
is probably good enough (tracking whether or not `None` is allowed for a particular trait is probably not possible).